src/UseCase/Transaction/RunTransaction.php

```diff
@@ -39,36 +44,105 @@ public function __construct(AdsClient $adsClient)
+        $command->setLastHash($account->getHash());
+        $command->setLastMsid($account->getMsid());
 
-        $getAccountResponse = $this->client->getAccount($address);
-        $account = $getAccountResponse->getAccount();
+        $response = $this->client->{$type}($command, true);
 
-        $command = CommandFactory::create($type, $params);
+        /** @var Tx $tx */
+        $tx = $response->getTx();
 
+        $commandResponse = new CommandResponse(
+            $address,
+            $tx->getData(),
+            $tx->getFee(),
+            $account->getHash(),
+            $account->getMsid()
+        );
+
+        $commandResponse->setTime($tx->getTime());
+
+        return $commandResponse;
+    }
+
+    /**
+     * @param string $type
+     * @param string $address
+     * @param string $signature
+     * @param \DateTime $time
+     * @param array $params
+     * @throws UnsupportedTransactionException
+     * @throws TooLowBalanceException
+     * @throws TransactionCannotBeProceedException
+     * @return CommandResponse
+     */
+    public function run(
+        string $type,
+        string $address,
+        string $signature,
+        \DateTime $time,
+        array $params
+    ): CommandResponse {
+        $this->validate($type);
+        $account = $this->getAccount($address);
+
+        $command = CommandFactory::create($type, $params);
         $command->setSender($address);
         $command->setLastHash($account->getHash());
         $command->setLastMsid($account->getMsid());
+        $command->setTimestamp($time->getTimestamp());
+        $command->setSignature($signature);
+
+        try {
+            $response = $this->client->{$type}($command, false);
+        } catch (CommandException $ex) {
+            if ($ex->getCode() === CommandError::LOW_BALANCE) {
+                throw new TooLowBalanceException('To low balance on account.');
+            }
+
+            throw new TransactionCannotBeProceedException($ex->getMessage());
+        }
+
+        /** @var Tx $tx */
+        $tx = $response->getTx();
+
+        $commandResponse = new CommandResponse(
+            $address,
+            $tx->getData(),
+            $tx->getFee(),
+            $account->getHash(),
+            $account->getMsid()
+        );
+        $commandResponse->setTransactionId($tx->getId());
+
+        return $commandResponse;
+    }
+
+    private function validate(string $type): void
+    {
+        $methodName = ($type === 'sendOne' || $type === 'sendMany') ? 'runTransaction' : $type;
+
+        if (!method_exists($this->client, $methodName)) {
+            throw new UnsupportedTransactionException(sprintf('Unsupported transaction type: %s', $type));
+        }
+    }
 
-        $response = $this->client->{$type}($command, $isDry);
+    private function getAccount(string $address)
+    {
+        $getAccountResponse = $this->client->getAccount($address);
+        $account = $getAccountResponse->getAccount();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "$account". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1488)